### DISCUSSION TITLE
Client Portals

### DISCUSSION THREAD

**Discussion by luke-ransfield** [Budibase/budibase#6381] [General]
I'm interested in creating a client portal; I've had a poke-around the Accountant, Agency and Customer Account portal templates, but I've come to the conclusions you would need one portal app per client/account (due to user roles/permissions)?

I've read across the discussions and it looks like Budibase is working on Groups/Workspaces for user-permissions which might solve what I'm looking for, but I'm curious to know if its possible to create one portal-style app where many different clients can access and only see the information related to them?

For full-context we're a product company that supplies various manufacturers with manufacturing files, at the moment we send these via a sharepoint, dropbox or even by email alongside a PDF of whatever we need them to manufacturer (a work order instruction). I'm hoping to transition us to something a bit more organized. Ideally a manufacturing partner can login to the portal, see only the jobs assigned to them, have the instructions listed as part of the job and the associate files, and be able to do simple things like ask for file revisions, update us on the progress on the job, etc. 

Is this something I could currently do with Budibase, or something I could expect to do in the future, or something that's outside of Budibase scope?

**Comment by melohagan** [comment]
Hi @lukxfnz 

I think you're correct in that the best way forward is to have a separate app per client. That way you can ensure their data/access is segregated. 

It does make managing changes to the app a bit of a hassle, needing to do export/import (https://docs.budibase.com/docs/portal#export-app) from a "development" app. 
This would also only work with external datasources, as the internal database data is saved with the app export.
On the other hand, by having separate apps per client, you can build client specific features without affecting all other clients. It's not git though - there's no app forking/branching/merging functionality. 

When the user groups come out, it will make it more convenient to group multiple users to a single client app, but it doesn't effect the currently available functionality. 

Ultimately Budibase is not built as a SaaS first platform, but there are some workarounds if needed.

**Reply by luke-ransfield** [reply]
Thanks @melohagan, great answer ~ I'll look forward to seeing groups/workspaces implemented then.
